src/cpp/include/openvino/genai/speech_recognition/asr_pipeline.hpp

```diff
@@ -0,0 +1,31 @@
+// Copyright (C) 2023-2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include "utils.hpp"
+
+namespace ov {
+namespace genai {
+
+// forward declaration
+class ASRPipelineImplBase;
```

Comment:
what is about Whisper pipeline? Will it work with ASRPipeline? I expect -yes. Please add tests for this.

Reply:
Agree, `ASRPipeline` should work with Whisper model. Once we add `ASRPipeline` I expect whisper pipeline to be deprecated.